src/Nethermind/Nethermind.Merge.Gnosis/GnosisMergePlugin.cs

```diff
@@ -0,0 +1,150 @@
+using Nethermind.Merge.Plugin;
+using Nethermind.Consensus.Transactions;
+using Nethermind.Consensus.AuRa.Transactions;
+using Nethermind.Consensus.Producers;
+using Nethermind.Consensus.Processing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Consensus.AuRa.Contracts;
+using Nethermind.Consensus.AuRa.InitializationSteps;
+using Nethermind.Consensus.AuRa.Contracts.DataStore;
+using Nethermind.Core;
+using Nethermind.Db;
+using Nethermind.Blockchain;
+using Nethermind.Blockchain.Data;
+using System.Collections.Generic;
+using Nethermind.Consensus;
+using Nethermind.Consensus.AuRa.Config;
+using Nethermind.State;
+using Nethermind.Specs.ChainSpecStyle;
+using System.Threading.Tasks;
+using Nethermind.Api;
+using Nethermind.Api.Extensions;
+
+namespace Nethermind.Merge.Gnosis
+{
+    public class GnosisMergePlugin : MergePlugin, IInitializationPlugin
+    {
+        protected override bool MatchVariant(string? variant)
+        {
+            return variant != null && variant == "AuRa";
+        }
+
+        public override Task Init(INethermindApi nethermindApi)
+        {
+            return base.Init(nethermindApi).ContinueWith(t =>
+            {
+                if (_mergeConfig.Enabled && MatchVariant(_mergeConfig.Variant))
+                {
+                    ((AuRaNethermindApi)nethermindApi).PoSSwitcher = _poSSwitcher;
+                }
+            });
+        }
+
+        protected override ITxSource? CreateTxSource(IStateProvider stateProvider)
```

Comment:
This looks like a direct copy from StartBlockProducerAuRa, maybe refactor that to be static? Or callable? Or just instantiate it here and call? Maybe better not to have duplication?

Reply:
hmmm, I think there was something different, I'll take a closer look.